docs/r2/migrate.ipynb

```diff
@@ -0,0 +1,404 @@
+        "    -   Default writers do not (yet) propagate across the `@tf.function` execution boundary - they are only detected when the function is traced - so best practice is to call `writer.as_default()` within the function body, and to ensure that the writer object continues to exist as long as the `@tf.function` is being used\n",
+        "\n",
+        "1.  The \"step\" value must be passed into each op via a the `step` argument\n",
+        "    -   TensorBoard requires a step value to render the data as a time series\n",
+        "    -   Explicit passing is necessary because the global step from TF 1.x has been removed, so each op must know the desired step variable to read\n",
+        "    -   To reduce boilerplate, experimental support for registering a default step value is available as `tf.summary.experimental.set_step()`, but this is provisional functionality that may be changed without notice\n",
+        "\n",
+        "1.  Function signatures of individual summary ops have changed\n",
+        "    -   Return value is now a boolean (indicating if a summary was actually written)\n",
+        "    -   The second parameter name (if used) has changed from `tensor` to `data`\n",
+        "    -   The `collections` parameter has been removed; collections are TF 1.x only\n",
+        "    -   The `family` parameter has been removed; just use `tf.name_scope()`\n",
+        "\n",
+        "1.  [Only for legacy graph mode / session execution users]\n",
+        "    -   First initialize the writer with `v1.Session.run(writer.init())`\n",
+        "    -   Use `v1.summary.all_v2_summary_ops()` to get all TF 2.0 summary ops for the current graph, e.g. to execute them via `Session.run()`\n",
+        "    -   Flush the writer with `v1.Session.run(writer.flush())` and likewise for `close()`\n",
+        "\n",
+        "If your TF 1.x code was instead using `tf.contrib.summary` API, it's much more similar to the TF 2.0 API, so `tf_upgrade_v2` script will automate most of the migration steps (and emit warnings or errors for any usage that cannot be fully migrated).  For the most part it just rewrites the API calls to `tf.compat.v2.summary`; if you only need compatibility with TF 2.0+ you can drop the `compat.v2` and just reference it as `tf.summary`."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "1GUZRWSkW3ZC"
+      },
+      "source": [
+        "## Additional tips\n",
+        "\n",
+        "In addition to the critical areas above, some auxiliary aspects have also changed:\n",
+        "\n",
+        "*  Conditional recording (like \"log every 100 steps\") has a new look\n",
+        "    -   To control ops and associated code, wrap them in a regular if statement (which works in eager mode and in `[@tf.function` via autograph](https://www.tensorflow.org/alpha/guide/autograph)) or a `tf.cond`\n",
```

Comment:
Backticks around `@tf.function` are not quite right, the first one should be inside the link bracket otherwise the link is broken